posixlib/src/main/resources/scala-native/sys/socket.c

```diff
@@ -13,6 +14,26 @@ typedef SSIZE_T ssize_t;
 #else
 #include <netinet/in.h>
 #include <sys/socket.h>
+#if !(defined __STDC_VERSION__) || (__STDC_VERSION__ < 201112L)
+#ifndef SCALANATIVE_SUPPRESS_STRUCT_CHECK_WARNING
+#warning "Size and order of C structures are not checked when -std < c11."
+#endif
+#else
+// Posix defines the order and type of required fields. Size of fields
+// and any internal or tail padding are left unspecified. This section
+// verifies that the C and Scala Native definitons match in each compilation
+// environment.
+//
+// The first sockaddr field in C has had size 2 and no padding after it
+// since time immemorial. Chedk that the Scala Native has the same.
+
+_Static_assert(offsetof(struct scalanative_sockaddr, sa_data) == 2,
+               "Unexpected size: scalanative_sockaddr sa_family");
+
+_Static_assert(offsetof(struct scalanative_sockaddr, sa_data) ==
+                   offsetof(struct sockaddr, sa_data),
+               "offset mismatch: sockaddr sa_data");
+#endif
```

Comment:
I really don't have the knowledge to comment on static asserts in relation to struct alignment.

In my far past working with C, we would set a POSIX or C standard in the make file for compiling, forcing the C library to thread that path. I would think that we should be able to generate code that is compatible with a C library in regards to struct alignment and padding for a certain standard but I have yet to experiment and thread my way through that idea.

In a previous PR https://github.com/scala-native/scala-native/pull/2093, you had some checking of struct lengths. Is this similar?

Reply:
re: In a previous PR #2093, you had some checking of struct lengths. Is this similar?

Exactly, the same concept of competently using the C compiler is applicable in many 
places.  To me the potential benefits of decreased execution time,  decreased development time
with easier correctness is worth presenting and explains a second, but not a third time.

Reply:
Again, thank you for your time & detail, Let me describe more. I am trying for professional discourse
and to filter out any crankiness or frustration on my part.  Forgive me if I fail in that effort.

re: I would think that we should be able to generate code that is compatible with a C library in regards to struct alignment and padding for a certain standard

To the best of my knowledge, specifying either padding or/and alignment are off in the fringes of C & Posix
standards.  I want to say that none do, and in the main that is true.  I do vividly remember that network IPv6
structures were designed and specified on purpose to be aligned and, effectively, packed.  I would
have to check the sponsorship of that code to see how it relates to IEEE C (and/or ISO C and/or POSIX).
I think there are some network extensions in those somewhere but I can not quote chapter and verse.

To the best of my knowledge that is a soft assumption that a developer will (os) link with a library
which is known to be compatible with the include files specified to the compiler.  That is, I think
there is a soft correspondance:  e.g. "Use Clang .h files and link with any library which matches them.
If you get it wrong, that is on you".

The `_Static_assert()` of this PR assures that one of the following happens:

1)  it is reported that structure checking can not happen and the build fails (with
     a way to take the blade guards off and force compilation)

2)  the SN and supplied .h files match (to a precision of interest)

3)  any mismatch (to that degree of precision of interest) is reported.